packages/design-systems/src/components/composite/Datagrid/SearchFilter/useCustomFilter.ts

```diff
@@ -136,15 +142,58 @@ export const useCustomFilter = <TData>({
   const [filterRows, setFilterRows] =
     useState<FilterRowData[]>(initialFilterRows());
 
+  const [prevFilterRows, setPrevFilterRows] = useState<FilterRowData[]>([]);
+
+  const calcNumberOfSearchConditions = useCallback(() => {
+    const isCurrentStateValid = (state: FilterRowData) => {
+      return (
+        state.currentState.column &&
+        state.currentState.condition &&
+        state.currentState.value
+      );
+    };
+    const count = filterRows.reduce((acc, row, index) => {
+      // First row does not have jointCondition
+      if (index === 0) {
+        if (isCurrentStateValid(row)) {
+          return acc + 1;
+        }
+        return acc;
+      }
+      if (isCurrentStateValid(row) && row.currentState.jointCondition) {
+        return acc + 1;
+      }
+      return acc;
```

Comment:
これも元からだし好みかもですが
```suggestion
      // First row does not have jointCondition
      if (（index === 0 || row.currentState.jointCondition） && isCurrentStateValid(row)) {
          return acc + 1;
      }
      return acc;
```
のほうが読みやすいなという印象です

Reply:
isCurrentStateValidがindexありきならそもそも `isCurrentStateValid（state: FilterRowData, index: number）` でいいかも。

Reply:
higuchiさんのsuggestしてくれているやつを採用しましょう。

排他の分岐がifの条件だけに現れていた方がわかりやすいです。

Reply:
いただいた提案だと条件が少し抜けていたので、下記のように変更しました。
(index === 0 && isCurrentStateValid(row)) が抜けてました。
```
const calcNumberOfSearchConditions = useCallback(() => {
    const isCurrentStateValid = (state: FilterRowData) => {
      return (
        state.currentState.column &&
        state.currentState.condition &&
        state.currentState.value
      );
    };
    const count = filterRows.reduce((acc, row, index) => {
      // First row does not have jointCondition
      if (
        (index === 0 && isCurrentStateValid(row)) ||
        (row.currentState.jointCondition && isCurrentStateValid(row))
      ) {
        return acc + 1;
      }
      return acc;
    }, 0);
    return count;
  }, [filterRows]);
```

Reply:
以下の2つは論理的に同じ条件だと思います。
```ts
(index === 0 && isCurrentStateValid(row)) || (row.currentState.jointCondition && isCurrentStateValid(row))
```
```ts
（index === 0 || row.currentState.jointCondition） && isCurrentStateValid(row)
```

`isCurrentStateValid(row)` は `(index === 0 && isCurrentStateValid(row))` と 
 `(row.currentState.jointCondition && isCurrentStateValid(row))` のどちらでも必要になるのでくくりだした形に変形した結果です。

Reply:
あー、くくりだすカッコの位置を見間違えてました、、、
修正しております。